src/api-clients/wikimedia-commons-api-client.js

```diff
@@ -0,0 +1,44 @@
+import HttpClient from './http-client.js';
+import urlUtils from '../utils/url-utils.js';
+
+const WIKIMEDIA_COMMONS_API_URL = 'https://commons.wikimedia.org/w/api.php';
+
+export const FILE_TYPE = {
+  bitmap: 'bitmap',
+  drawing: 'drawing',
+  video: 'video',
+  audio: 'audio',
+  pdf: 'pdf'
+};
+
+class WikimediaCommonsApiClient {
+  static inject() { return [HttpClient]; }
+
+  constructor(httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  queryMediaFiles({ searchText, fileTypes = Object.values(FILE_TYPE), thumbnailHeight = 180, limit = 50, offset = 0 }) {
+    const queryString = urlUtils.composeQueryString({
+      action: 'query',
+      format: 'json',
+      uselang: 'en',
+      generator: 'search',
+      gsrsearch: `filetype:${fileTypes.join('|')} ${searchText}`,
+      gsrlimit: limit,
+      gsroffset: offset,
+      gsrnamespace: '*',
+      prop: 'info|imageinfo',
+      inprop: 'url',
+      iiprop: 'url|size|mime',
+      iiurlheight: thumbnailHeight,
+      origin: '*'
+    });
+
+    return this.httpClient
+      .get(`${WIKIMEDIA_COMMONS_API_URL}?${queryString}`, { responseType: 'json' })
```

Comment:
I'm thinking we should url encode the `queryString` part due to `searchText`, or?

Reply:
`urlUtils.composeQueryString` returns an encoded, ready to use query string.

Reply:
Indeed it does 😬